moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp

```diff
@@ -335,6 +363,13 @@ bool PlanningPipeline::generatePlan(const planning_scene::PlanningSceneConstPtr&
         RCLCPP_INFO(node_->get_logger(), "Calling PlanningResponseAdapter '%s'", res_adapter->getDescription().c_str());
         res_adapter->adapt(planning_scene, mutable_request, res);
         publishPipelineState(mutable_request, res, res_adapter->getDescription());
+
+        // check for timeout
```

Comment:
Maybe I'm missing it, but it looks like we will hit a timeout here if the planners use up the allowed planning time. Optimizing planners usually do that, so I think we should either ignore response adapters or use a separate budget

Reply:
Yeah I think the ultimate solution to this is to give user more control over the timeout. Maybe let them decide timeout of each section (and each planner) separately? But that would require changing the `MotionPlanRequest` message so I'm not sure. What's your suggestion on this? Should we make the PR more complex or just ignore the response adapters for now?

Reply:
Can we ensure for now that a fraction of the time will be reserved for the planners? Something like 100ms in any case?

Reply:
But in that case the planner might still use up all the reserved time. Or are you saying that giving the request adapter at least 100ms?

Reply:
the response adapter should have a reserved time budget which is deducted from the planner's allowed planning time.

Reply:
hi, I add an extra check to ensure that the response adapter has 10ms left in the latest commit